src/Omega_h_macros.h

```diff
@@ -90,4 +90,8 @@
 #define OMEGA_H_DLL
 #endif
 
+#if !defined(__HIP_DEVICE_COMPILE__) && !defined(__CUDA_ARCH__)
```

Comment:
Please add the following to support the sycl backend `&& !defined(__SYCL_DEVICE_ONLY__)`. 

The SYCL 2020 standard says https://registry.khronos.org/SYCL/specs/sycl-2020/pdf/sycl-2020.pdf:
> ` __SYCL_DEVICE_ONLY__` is defined to 1 if the source file is being compiled with a SYCL device compiler which does not produce host binary;